pkg/cmd/release-payload-controller/batchjob_controller.go

```diff
@@ -0,0 +1,279 @@
+	batchv1listers "k8s.io/client-go/listers/batch/v1"
+	"reflect"
+	"time"
+
+	utilruntime "k8s.io/apimachinery/pkg/util/runtime"
+	"k8s.io/apimachinery/pkg/util/wait"
+	"k8s.io/client-go/tools/cache"
+	"k8s.io/client-go/util/workqueue"
+	"k8s.io/klog/v2"
+
+	"github.com/openshift/library-go/pkg/operator/events"
+)
+
+const (
+	// ReleaseCreationJobUnknownMessage release creation job unknown message
+	ReleaseCreationJobUnknownMessage = "Release creation job unknown"
+
+	// ReleaseCreationJobPendingMessage release creation job pending message
+	ReleaseCreationJobPendingMessage = "Release creation job pending"
+
+	// ReleaseCreationJobFailureMessage release creation job failure message
+	ReleaseCreationJobFailureMessage = "Release creation Job failed"
+
+	// ReleaseCreationJobSuccessMessage release creation job success message
+	ReleaseCreationJobSuccessMessage = "Release creation Job completed"
+)
+
+// BatchJobStatusController is responsible for watching batchv1.Jobs, in the job-namespace, and
+// updating the respective ReleasePayload with the status, of the job, when it completes.
+// The BatchJobStatusController watches for changes to the following resources:
+//   - batchv1.Jobs
+// The reads the following pieces of information:
+//   - .status.releaseCreationJobResult.coordinates.name
+//   - .status.releaseCreationJobResult.coordinates.namespace
+// and write the following information:
+//   - .status.releaseCreationJobResult.status
+type BatchJobStatusController struct {
+	releasePayloadNamespace string
+	releasePayloadLister    releasepayloadlister.ReleasePayloadLister
+	releasePayloadClient    releasepayloadclient.ReleaseV1alpha1Interface
+
+	batchJobNamespace string
+	batchJobLister    batchv1listers.JobLister
+
+	eventRecorder events.Recorder
+
+	cachesToSync []cache.InformerSynced
+
+	queue workqueue.RateLimitingInterface
+}
+
+func NewBatchJobStatusController(
+	releasePayloadNamespace string,
+	releasePayloadInformer releasepayloadinformer.ReleasePayloadInformer,
+	releasePayloadClient releasepayloadclient.ReleaseV1alpha1Interface,
+	batchJobNamespace string,
+	batchJobInformer batchv1informers.JobInformer,
+	eventRecorder events.Recorder,
+) (*BatchJobStatusController, error) {
+	c := &BatchJobStatusController{
+		releasePayloadNamespace: releasePayloadNamespace,
+		releasePayloadLister:    releasePayloadInformer.Lister(),
+		releasePayloadClient:    releasePayloadClient,
+		batchJobNamespace:       batchJobNamespace,
+		batchJobLister:          batchJobInformer.Lister(),
+		eventRecorder:           eventRecorder.WithComponentSuffix("batchjob-status-controller"),
+		queue:                   workqueue.NewNamedRateLimitingQueue(workqueue.DefaultControllerRateLimiter(), "BatchJobStatusController"),
+	}
+
+	c.cachesToSync = append(c.cachesToSync, releasePayloadInformer.Informer().HasSynced)
+	c.cachesToSync = append(c.cachesToSync, batchJobInformer.Informer().HasSynced)
+
+	batchJobFilter := func(obj interface{}) bool {
```

Comment:
you may create a releasePayloadFilter because once status is successful you never re-inspect, right?

Reply:
Done!